src/auth/models/strategy.class.ts

```diff
@@ -3,15 +3,24 @@ import * as express from 'express'
 import { NextFunction, Request, RequestHandler, Response } from 'express'
 import passport from 'passport'
 import { AUTH } from '../auth.constants'
+import { OAuth2Metadata } from '../oauth2'
+import { OpenIDMetadata } from '../oidc'
+import { FileSessionMetadata, RedisSessionMetadata } from '../session/models/sessionMetadata.interface'
 
-export class Authentication extends events.EventEmitter {
-    protected readonly strategyName: string
+export abstract class Strategy extends events.EventEmitter {
+    public readonly strategyName: string
     protected readonly router = express.Router({ mergeParams: true })
```

Comment:
could we just tidy this up a bit? we have 2 imports from express, 
```
import * as express
import { NextFunction, Request, RequestHandler, Response } from 'express'
```

Think I would prefer if we only import what we need, e.g.
```
import { NextFunction, Request, RequestHandler, Response, Router } from 'express'
```

Reply:
Agreed, done